floris/optimization/load_optimization/load_optimization.py

```diff
@@ -0,0 +1,459 @@
+    exp_thrust: float = 1.0,
+):
+    """Find the value of A that satisfies the target revenue to VOC ratio.
+
+    Args:
+        fmodel (FlorisModel): FlorisModel object
+        target_rev_voc_ratio (float): Target revenue to VOC ratio
+        load_ambient_tis (list or np.array): Ambient 'load' turbulence intensity for each findex
+        wake_slope (float, optional): Wake slope. Defaults to 0.3.
+        max_dist_D (flat, optional): Maximum distance in rotor diameters. Defaults to 10.0.
+        exp_ws_std (float, optional): Exponent for the wind speed standard deviation.
+            Defaults to 1.0.
+        exp_thrust (float, optional): Exponent for the thrust. Defaults to 1.
+
+    Returns:
+        float: Value of A that satisfies the target revenue to VOC ratio
+
+    """
+
+    # Compute farm revenue
+    farm_revenue = compute_farm_revenue(
+        fmodel=fmodel,
+    )
+
+    # Compute farm VOC
+    farm_voc = compute_farm_voc(
+        fmodel=fmodel,
+        A=1.0,
+        load_ambient_tis=load_ambient_tis,
+        wake_slope=wake_slope,
+        max_dist_D=max_dist_D,
+        exp_ws_std=exp_ws_std,
+        exp_thrust=exp_thrust,
+    )
+
+    return (farm_revenue.sum() / farm_voc.sum()) / target_rev_voc_ratio
+
+
+def optimize_derate(
+    fmodel: FlorisModel,
+    A: float,
+    load_ambient_tis: np.array,
+    wake_slope: float = 0.3,
+    max_dist_D: float = 10.0,
+    exp_ws_std: float = 1.0,
+    exp_thrust: float = 1.0,
+    initial_power_setpoint: np.array = None,
+    derating_levels: np.array = np.linspace(1.0, 0.001, 5),
+):
+    """Optimize the derating of each turbine to maximize net revenue.
+
+    Args:
+        fmodel (FlorisModel): FlorisModel object
+        A (float): Coefficient for the VOC calculation
+        load_ambient_tis (list or np.array): Ambient 'load' turbulence intensity for each findex
+        wake_slope (float, optional): Wake slope. Defaults to 0.3.
+        max_dist_D (flat, optional): Maximum distance in rotor diameters. Defaults to 10.0.
+        exp_ws_std (float, optional): Exponent for the wind speed standard deviation.
+            Defaults to 1.0.
+        exp_thrust (float, optional): Exponent for the thrust. Defaults to 1.
+        initial_power_setpoint (np.array, optional): Initial power setpoint for each turbine.
+            Defaults to None.
+        derating_levels (np.array, optional): Array of derating levels to consider.
+            Defaults to np.linspace(1.0, 0.001, 5).
+
+    """
+
+    # Ensure we're in derating mode
+    fmodel.set_operation_model("simple-derating")
+
+    # If initial set point not provided, set to rated (assumed max) power
+    if initial_power_setpoint is None:
+        max_power = get_max_powers(fmodel)
+        initial_power_setpoint = np.tile(max_power, (fmodel.n_findex, 1))
+
+    # Initialize the the test power setpoints
+    test_power_setpoint = initial_power_setpoint.copy()
+    final_power_setpoint = initial_power_setpoint.copy()
```

Comment:
could `final_power_setpoint` be renamed `optimal_power_setpoint` or `power_setpoint_opt`? "Final" isn't super clear to me

Reply:
I've now added this